lib/nan_test/lfric/nan_test.jinja

```diff
@@ -0,0 +1,211 @@
+                                              real32, real64, &
+                                              stderr=>Error_Unit
+    use field_mod, only : field_type
+    use nan_test_base_mod, only : NANTestBaseType, is_enabled
+
+    implicit none
+
+    type, extends(NANTestBaseType), public:: nan_test_PSyDataType
+
+    contains
+        ! The LFRic-specific procedures defined here
+        procedure :: DeclareFieldDouble,  ProvideFieldDouble
+        procedure :: DeclareFieldVectorDouble,  ProvideFieldVectorDouble
+
+        {# Collect the various procedures for the same generic interface #}
+        {# ------------------------------------------------------------- #}
+        {% set all_declares=[] %}
+        {% set all_provides=[] %}
+        {% for name, type, bits in ALL_TYPES %}
+          {{ all_declares.append("DeclareScalar"~name) or "" -}}
+          {{ all_provides.append("ProvideScalar"~name) or "" -}}
+          {% for dim in ALL_DIMS %}
+            {{ all_declares.append("DeclareArray"~dim~"d"~name) or "" -}}
+            {{ all_provides.append("ProvideArray"~dim~"d"~name) or "" -}}
+          {% endfor %}
+        {% endfor %}
+
+        {% set indent="            " %}
+        ! Declare generic interface for PreDeclareVariable:
+        generic, public :: PreDeclareVariable => &
+            DeclareFieldDouble, &
+            DeclareFieldVectorDouble, &
+            {{all_declares|join(", &\n"+indent) }}
+
+        !> The generic interface for providing the value of variables
+        !! (which checks for non normal IEEE numbers)
+        generic, public :: ProvideVariable => &
+            ProvideFieldDouble,       &
+            ProvideFieldVectorDouble, &
+            {{all_provides|join(", &\n"+indent) }}
+                                              
+    end type nan_test_PSyDataType
+
+Contains
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine does not do anything (as declaration is not needed
+    !!for NAN checking).
+    !! @param[inout] this The instance of the nan_test_PSyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    !! @param[inout] this The instance of the nan_test_PSyDataType.
+    subroutine DeclareFieldDouble(this, name, value)
+        implicit none
+        class(nan_test_PSyDataType), intent(inout), target :: this
+        character(*), intent(in) :: name
+        type(field_type), intent(in) :: value
+
+    end subroutine DeclareFieldDouble
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine checks whether an LFRic field has NAN or infinite
+    !! floating point values.
+    !! @param[inout] this The instance of the nan_test_PSyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    subroutine ProvideFieldDouble(this, name, value)
```

Comment:
As with the netcdf extraction lib, should this just be `ProvideField` rather than `ProvideFieldDouble` - especially since the implementation is not type dependent?

Reply:
Indeed.